File: frontend/GameHubFront/src/Games/batalhaRural/services/validators/InputValidator.ts
import { Board } from "../../entities/Board";
import { TokenInfo } from "../../helpers/TokenInfo";
import { Logger } from "../../helpers/Logger";

export class InputValidator {

    static isInputValid(tokenInfo: TokenInfo, board: Board): boolean {
        const boardLength = board.tiles.length;

        // verificando se as coordenadas estão dentro do tabuleiro
        if ((tokenInfo.baseOnePosX > boardLength) || (tokenInfo.baseOnePosY > boardLength) || (tokenInfo.baseOnePosX < 1) || (tokenInfo.baseOnePosY < 1)) {
            Logger.invalidInput();

            return false;
        }

        return true;
    }
}

File: backend/gameHubBack/services/BatalhaRural/Validators/InputValidator.cs
namespace gameHubBack.Services.BatalhaRural.Validators;

public static class InputValidator
{
    public static bool IsInputValid(TokenInfo tokenInfo, string[][] tiles)
    {
        var boardLength = tiles.Length;

        if (tokenInfo.BaseOnePosX > boardLength || tokenInfo.BaseOnePosY > boardLength ||
            tokenInfo.BaseOnePosX < 1 || tokenInfo.BaseOnePosY < 1)
        {
            return false;
        }

        return true;
    }
}